src/components/autocomplete/js/autocompleteController.js

```diff
@@ -64,6 +66,18 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
     $mdUtil.initOptionalProperties($scope, $attrs, { searchText: null, selectedItem: null });
     $mdTheming($element);
     configureWatchers();
+
+    if ($scope.componentId) {
+      removeFromRegistry = $mdComponentRegistry.register({
+        // Restricted API interface for the component registry.
+        // We don't want to provide the whole controller as API.
+        clearCache: function() {
+          // Run the clearCache method in the controller context.
+          ctrl.clearCache();
```

Comment:
Had that before, and it worked, but then I won't be able to test that in jasmine, because the spy can't access this one. But I think, I better change that back and remove that particular test. The behavior of the component registry is covered in other tests too, so it will be still fine.